pkg/k8shandler/status.go

```diff
@@ -47,15 +45,50 @@ func updateNodeStatus(node *nodeState, dpl *v1alpha1.ElasticsearchStatus) {
 	if node.Actual.StatefulSet != nil {
 		nodeStatus.StatefulSetName = node.Actual.StatefulSet.Name
 	}
+
+	if node.Desired.Roles != nil {
+		nodeStatus.Roles = node.Desired.Roles
+	}
 	dpl.Nodes = append(dpl.Nodes, nodeStatus)
 }
 
-func AddStatusMessage(message string, dpl *v1alpha1.Elasticsearch) error {
-	dpl.Status.Message = message
-	err := sdk.Update(dpl)
+func clusterHealth(dpl *v1alpha1.Elasticsearch) string {
+	if len(dpl.Status.Nodes) == 0 {
+		return healthUnknown
+	}
+
+	podName := dpl.Status.Nodes[0].PodName
+
+	config := &ExecConfig{
+		namespace:      dpl.ObjectMeta.Namespace,
+		podName:        podName,
+		containerName:  "elasticsearch",
+		command:        []string{"es_util", "--query=_cluster/health?pretty=true"},
+		kubeConfigPath: LookupEnvWithDefault("KUBERNETES_CONFIG", "/etc/origin/master/admin.kubeconfig"),
+		masterURL:      "https://kubernetes.default.svc",
+		stdOut:         true,
+		stdErr:         true,
+		tty:            false,
+	}
+
+	execOut, _, err := PodExec(config)
 	if err != nil {
-		return fmt.Errorf("failed to update Elasticsearch status: %v", err)
+		logrus.Debug(err)
+		return healthUnknown
 	}
 
-	return nil
+	var result map[string]interface{}
+
+	err = json.Unmarshal(execOut.Bytes(), &result)
+	if err != nil {
+		logrus.Debug("could not unmarshal: %v", err)
+		return healthUnknown
+
+	}
+	if _, present := result["status"]; !present {
+		logrus.Debug("response from elasticsearch health API did not contain 'status' field")
```

Comment:
I think this may be more severe than just a `Debug` message. If we don't see a `status` field does that mean ES is non responsive?

Reply:
No, I forgot to change it. Good catch ;)